sematic/types/type.py

```diff
@@ -1,7 +1,14 @@
 # Standard Library
 import abc
+import sys
 import typing
 
+if (sys.version_info.major, sys.version_info.minor) >= (3, 10):
+    # Standard Library
+    from types import UnionType  # type: ignore
+else:
+    UnionType = typing.get_origin(typing.Union[int, float])
```

Comment:
Is there no way to extract the origin of `typing.Union` without dummy-parametrizing it with `[int, float]`?
Please add code comment.

Reply:
```suggestion
    # Different types from the typing module have different behaviors
    # when getting origin--behaviors which differ with python version.
    # Rather than hard-coding what the origin type is when you have
    # parameterized a Union, we just use an example. Why use two types
    # instead of just Union[int]? Because that is disallowed--all Union
    # parameterizations must have at least two types.
    UnionType = typing.get_origin(typing.Union[int, float])
```

Reply:
added a coment!